### PR TITLE
DOC: fixing GL08 errors for pandas.IntervalIndex.left, pandas.IntervalIndex.mid and pandas.IntervalIndex.length

### PR DESCRIPTION
**Pull request by j36426052**
All GL08 Errors resolved in the following cases:

scripts/validate_docstrings.py --format=actions --errors=GL08 pandas.IntervalIndex.left
scripts/validate_docstrings.py --format=actions --errors=GL08 pandas.IntervalIndex.mid
scripts/validate_docstrings.py --format=actions --errors=GL08 pandas.IntervalIndex.length

xref #57443

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -827,18 +827,79 @@ def _is_comparable_dtype(self, dtype: DtypeObj) -> bool:
...
+        See Also
+        --------
+        IntervalIndex : The structure of IntervalIndex.
+
+        Examples
+        --------
+        >>> pd.interval_range(start=0, end=5)
+        IntervalIndex([(0, 1], (1, 2], (2, 3], (3, 4], (4, 5]], dtype='interval[int64, right]')
+        >>> pd.interval_range(start=0, end=5).length
+        Int64Index([1, 1, 1, 1, 1], dtype='int64')
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:901]:
```suggestion
        Index([1, 1, 1, 1, 1], dtype='int64')
```

File: pandas/core/indexes/interval.py [hunk 2]
```diff
@@ -827,18 +827,79 @@ def _is_comparable_dtype(self, dtype: DtypeObj) -> bool:
...
+        See Also
+        --------
+        IntervalIndex : The structure of IntervalIndex.
+
+        Examples
+        --------
+        >>> pd.interval_range(start=0, end=5)
+        IntervalIndex([(0, 1], (1, 2], (2, 3], (3, 4], (4, 5]], dtype='interval[int64, right]')
+        >>> pd.interval_range(start=0, end=5).left
+        Int64Index([0, 1, 2, 3, 4], dtype='int64')
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:848]:
```suggestion
        Index([0, 1, 2, 3, 4], dtype='int64')
```

File: pandas/core/indexes/interval.py [hunk 3]
```diff
@@ -827,18 +827,79 @@ def _is_comparable_dtype(self, dtype: DtypeObj) -> bool:
...
+        """
         return Index(self._data.left, copy=False)
 
+
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:852]:
```suggestion
```

**Comment by j36426052** [comment]
I want to know that what should I do next ? I am not sure what should I fix the error, because some of it I'm not really understand. Or should I just skip it?

**Comment by j36426052** [comment]
pre-commit.ci autofix

File: pandas/core/indexes/interval.py [hunk 4]
```diff
@@ -835,10 +857,54 @@ def right(self) -> Index:
...
     @cache_readonly
     def mid(self) -> Index:
+        """
+        Return the intervals' mean by Float64Index.
+
+        Return Float64Index which contains all intervals' left and right value mean.
+
+        Returns
+        -------
+        Float64Index
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:867]:
```suggestion
        Return the midpoint of each interval.

        Returns
        -------
        Index
```

File: pandas/core/indexes/interval.py [hunk 5]
```diff
@@ -835,10 +857,54 @@ def right(self) -> Index:
...
     def length(self) -> Index:
+        """
+        Return interval's length.
+
+        Return number index which contains all intervals' length value
+        in Intervalindex's interval.
+
+        Returns
+        -------
+        Int64Index or Float64Index : depend on if the type of length value
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:894]:
```suggestion
        Return the intervals' length.

        Returns
        -------
        Index
```

**Review by datapythonista — COMMENTED** [review]
I think those are good candidates to reuse a docstring, together with `right`. You can see how we do it in many places with the `@doc` decorator.

Some general comments:

- For `See Also` we want for `left` to surely have `right`. I think we actually want to refer the rest of the 4 methods to each other.
- In `Return` besides the type, in the next line we want to have a description of what's being returned.
- For the `Examples` we want to assign the interval index to a variable, and then show the method, not to create one instance, and then create it again with the method or property being called.

**Comment by mroeschke** [comment]
@datapythonista just FYI there's an effort to minimize/remove docstring reuse (https://github.com/pandas-dev/pandas/pull/57683, https://github.com/pandas-dev/pandas/issues/57578) to minimize complexity of the docstring reuse mechanisms

**Comment by datapythonista** [comment]
@j36426052 do you want to continue working in this PR? Besides addressing the comments you'll have to resolve the conflicts.

**Comment by j36426052** [comment]
Sorry that I'm not really sure what part I need to fix. The comment part, should I reuse some comment to let it's structure more complete. I'm not sure how to minimize/remove docstring. 

Also the comflict in Numpy Dev, it show "test-data.xml not found", but I don't see some document talk about this.

Or should I just skip the above I mention. Maybe you mean other comflict?

**Comment by Aloqeely** [comment]
As mroeschke said, there's no need to worry about reusing docstrings as they are trying to minimize docstring reuse.
Focus on applying the 3 suggested "general comments", then make sure the checks pass.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**